packages/docusaurus-theme-classic/src/theme/CodeBlock/WordWrapButton/index.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useCallback, useState, useEffect} from 'react';
+import clsx from 'clsx';
+import {translate} from '@docusaurus/Translate';
+import type {Props} from '@theme/CodeBlock/WordWrapButton';
+
+import styles from './styles.module.css';
+
+export default function WordWrapButton({
+  codeBlockRef,
+  className,
+}: Props): JSX.Element | null {
+  const [isEnabled, setIsEnabled] = useState(false);
+  const [isCodeScrollable, setIsCodeScrollable] = useState<boolean>(false);
+  const toggleWordWrapCode = useCallback(() => {
+    setIsEnabled((value) => !value);
+
+    const codeElement = codeBlockRef.current!.querySelector('code');
+    codeElement?.classList.toggle(styles.codeWithWordWrap!);
+  }, [codeBlockRef]);
+  const updateCodeIsScrollable = useCallback(() => {
+    const {scrollWidth, clientWidth} = codeBlockRef.current!;
+    setIsCodeScrollable(scrollWidth > clientWidth);
+  }, [codeBlockRef]);
+  const title = translate({
+    id: 'theme.CodeBlock.wordWrapToggle',
+    message: 'Toggle word wrap',
+    description:
+      'The title attribute for toggle word wrapping button of code block lines',
+  });
+
+  useEffect(() => {
+    updateCodeIsScrollable();
+
+    window.addEventListener('resize', updateCodeIsScrollable);
+
+    return () => {
+      window.removeEventListener('resize', updateCodeIsScrollable);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!isCodeScrollable) {
```

Comment:
Little issue: 
- mobile
- enable word wrap
- resize

Now `isCodeScrollable = false` and the button disappears => no way to turn word wrap off

<img width="485" alt="image" src="https://user-images.githubusercontent.com/749374/162227760-71e1d75d-1f12-4511-b9ec-070c9fbb92d7.png">